### PR TITLE
Restructure and add information about I18n

### PR DESCRIPTION
**Pull request by llewelld**
Restructures the sections on Internationalisation to distinguish more clearly between Platform and Application configurations. Adds additional information (largely taken from our internal wiki) on best practice for how to handle internationalisation on Sailfish OS.

### CODE REVIEW

File: Reference/I18n/Platform/README.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+%files
+%{_datadir}/translations/*.qm
+
+(...)
+
+%package ts-devel
+Summary: Translation source for %{name}
+License: XXX
+Group: XXX
```
Comment (martyone) [inline Reference/I18n/Platform/README.md:28]: The Group tag should be probably omitted.

File: Reference/I18n/README.md [hunk 2]
```diff
@@ -1,82 +1,34 @@
 ---
-title: I18n
+title: Internationalisation
```
Comment (martyone) [inline Reference/I18n/README.md:2]: I think the policy was to keep title and path aligned, so we should move files too.

File: Reference/I18n/Platform/README.md (hunk 1 above)
Comment (llewelld) [reply Reference/I18n/Platform/README.md]: Thanks; removed.

File: Reference/I18n/README.md (hunk 2 above)
Comment (llewelld) [reply Reference/I18n/README.md]: In that case I've switched it back to `I18n` to avoid breaking inward links.

Comment (llewelld) [reply Reference/I18n/README.md]: I'll need to change the directory names for the sub pages as well...

Comment (llewelld) [reply Reference/I18n/README.md]: ...done.

File: Reference/I18n/Platform_Configuration/README.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    Q_OBJECT
+public:
+    AppTranslator(QObject *parent)
+        : QTranslator(parent)
+    {
+        qApp->installTranslator(this);
+    }
+
+    virtual ~AppTranslator()
+    {
```
Comment (Thaodan) [inline Reference/I18n/Platform_Configuration/README.md:159]: I think this should be using override.

File: Reference/I18n/I18n_Conventions/README.md [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+## Colons
+
+Colons are not carried in translations. If colons are needed, those are added in code level. Note e.g. combo boxes do not use colon. Colon is a style choice and the colour already differentiates the two strings.
+
+## Directionality
+
+Some locales have specific requirements for directionality of content. For example, in English-speaking locales, text and content should be laid out left-to-right and top-to-bottom, however in some other locales a different directionality is required (e.g., top-to-bottom then left-to-right, or right-to-left). Some content will also need to be mirrored (e.g., directional arrow images) when the layout is reversed.
+
+Applications should be written so that the layout is internationalised, by making use of the directionality support in Qt Quick, via the `horizontalAlignment`, `layoutMirroring`, `layoutDirection`, and `effectiveLayoutDirection` properties, and querying `Qt.application.layoutDirection` to determine which alignment to use at run-time (the value of which is determined from the layout direction defined in the currently active translation file).
+
```
Comment (pvuorela) [inline Reference/I18n/I18n_Conventions/README.md:76]: We haven't really done polished RTL support and when we do I'm thinking we probably have the same layout to some extend. Could just drop this part here, revisit later if/when need.

File: Reference/I18n/I18n_Conventions/README.md [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+## Application names
+
+When creating a `.desktop` file for an application, localised names can be supported as follows:
+
+```
+Name=Appname
+X-MeeGo-Logical-Id=appname-ap-name
+X-MeeGo-Translation-Catalog=appname
```
Comment (pvuorela) [inline Reference/I18n/I18n_Conventions/README.md:84]: This is the legacy naming, X-Amber prefix support now in mlite.

File: Reference/I18n/I18n_Conventions/README.md [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+
+Time localisation patterns define:
+
+ * The punctuation marks (e.g. from "h:mm" to "h.mm").
+ * Changes between the expressions `h`/`hh` and `m`/`mm` according to the leading zeros usage
+   in the language. 
+
+The Sailfish Silica project adds separate strings for short and long names of months and long
+names of days of the week in their basic forms. Translating these strings is done using the
+capitalisation they would use in the middle of a time/date representation.
```
Comment (pvuorela) [inline Reference/I18n/I18n_Conventions/README.md:181]: Another Silica translation specific thing which I'm not sure if should belong here.

File: Reference/I18n/Platform_Configuration/README.md (hunk 3 above)
Comment (llewelld) [reply Reference/I18n/Platform_Configuration/README.md:159]: Thanks; changed.

File: Reference/I18n/I18n_Conventions/README.md [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+|  he  |  header                 | Header in view, e.g. Media player: Music                |
+|  bt  |  button                 | Application area buttons                                |
+|  no  |  notification           | Notification                                            |
+|  bu  |  button                 | Notification area buttons                               |
+|  ba  |  banner                 | Event banner, e.g. 1 received SMS                       |
+|  me  |  menu item              | Pull-down or push-up menu item, e.g. Edit contact       |
+|  li  |  list item              | Item without data to be displayed, e.g. (unnamed)       |
+|  la  |  label                  | Context menu label, e.g. First name                     |
+|  va  |  value                  | Context menu value, e.g. In Settings: Language: Finnish |
+|  ph  |  textfield placeholders |
```
[the remark this answers is not in the document]
Comment (llewelld) [reply Reference/I18n/I18n_Conventions/README.md:60]: I've removed `ba` specifically and added a comment about the looseness of the guidelines. I'm not sure which others should be dropped if any.

File: Reference/I18n/I18n_Conventions/README.md [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+
+## Line breaks in translations
+
+It is recommended not to use line breaks in user interface translations as typically translations are 1-2 sentences long and line break can cause text being badly aligned. Exceptions like several paragraphs of text: Use \<\p> or \<br\> for line break.
+
+## Colons
+
+Colons are not carried in translations. If colons are needed, those are added in code level. Note e.g. combo boxes do not use colon. Colon is a style choice and the colour already differentiates the two strings.
```
[the remark this answers is not in the document]
Comment (llewelld) [reply Reference/I18n/I18n_Conventions/README.md]: I've reworded this section to hopefully make it clearer.

File: Reference/I18n/I18n_Conventions/README.md (hunk 4 above)
Comment (llewelld) [reply Reference/I18n/I18n_Conventions/README.md:75]: I've removed this section. I recall there was some work to support RTL for Arabic done by [community member mad_dev](https://forum.sailfishos.org/t/unofficial-rtl-support/671), but I guess there's still a lot to be done there.

File: Reference/I18n/I18n_Conventions/README.md (hunk 5 above)
Comment (llewelld) [reply Reference/I18n/I18n_Conventions/README.md]: Amended.

File: Reference/I18n/I18n_Conventions/README.md [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+| `dd`   | the day as number with a leading zero (01 to 31).                                          |
+| `ddd`  | the abbreviated localised day name (e.g. 'Mon' to 'Sun'). Uses QDate::shortDayName().      |
+| `dddd` | the long localised day name (e.g. 'Monday' to 'Sunday'). Uses QDate::longDayName().        |
+| `M`    | the month as number without a leading zero (1-12).                                         |
+| `MM`   | the month as number with a leading zero (01-12).                                           |
+| `MMM`  | the abbreviated localised month name (e.g. 'Jan' to 'Dec'). Uses QDate::shortMonthName().  |
+| `MMMM` | the long localised month name (e.g. 'January' to 'December'). Uses QDate::longMonthName(). |
+| `yy`   | the year as two digit number (00-99).                                                      |
+| `yyyy` | the year as four digit number.                                                             |
+| `cccc` | the wide stand-alone form for the day of the week (e.g. 'Monday'). Can be used only with DateFull and DateLong. |
```
[the remark this answers is not in the document]
Comment (llewelld) [reply Reference/I18n/I18n_Conventions/README.md]: I've removed this section.

File: Reference/I18n/I18n_Conventions/README.md (hunk 6 above)
Comment (llewelld) [reply Reference/I18n/I18n_Conventions/README.md]: This is also from the removed section. Thanks for helping straighten this out.

File: Reference/I18n/I18n_Conventions/README.md [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+    onClicked: Qt.quit()
+}
+```
+
+Lines starting with `//:` indicate context, provided to the translator to help them in translating. These are always worth adding and are helpful to the translators, but not compulsory.
+
+The following line starting with `//%` indicates the engineering English, a fallback translation if proper translation is missing.
+Last creates the actual translation. The ID is formed as `<component>-<placement>-<descriptive text>`.
+
+When developing platform components, `qtTrId()` and `qsTrId()` *must* be used. If you're developing your own application we also recommend that you follow the same convention, but you can of course also use `tr()` and `qsTr()` respectively if you prefer.
```
Comment (pvuorela) [inline Reference/I18n/I18n_Conventions/README.md:39]: I'm not actually sure which one to recommend for third party apps. tr() type of translation is the more common way.

**Review by pvuorela — APPROVED** [review]
One comment whether we should suggest third party app qsTrId vs qsTr but generally good. Approving already.

File: Reference/I18n/I18n_Conventions/README.md (hunk 10 above)
Comment (llewelld) [reply Reference/I18n/I18n_Conventions/README.md:39]: In the past I've started off using `tr()` in apps and then gone on to regret it and had to spend effort switching to using the more flexible ids. But maybe this is a personal thing.

Comment (llewelld) [reply Reference/I18n/I18n_Conventions/README.md]: I've made some small changes to make it less opinionated for the app developer case.

**Comment by llewelld** [comment]
Rebased; normalised use of `.spec` and `.yaml`.

**Pull request merged**